frontends/concrete-python/tests/compilation/test_circuit.py

```diff
@@ -512,11 +513,91 @@ def test_circuit_compile_sim_only(helpers):
 
     configuration = helpers.configuration()
 
-    @compiler({"x": "encrypted", "y": "encrypted"})
+    @fhe.compiler({"x": "encrypted", "y": "encrypted"})
     def f(x, y):
         lut = LookupTable(list(range(128)))
         return lut[x + y]
 
     inputset = [(np.random.randint(0, 2**4), np.random.randint(0, 2**5)) for _ in range(2)]
     circuit = f.compile(inputset, configuration.fork(fhe_simulation=True, fhe_execution=False))
     assert f(*inputset[0]) == circuit.simulate(*inputset[0])
+
+
+@pytest.mark.parametrize(
+    "function,parameters,expected_statistics",
+    [
+        pytest.param(
+            lambda x: x**2,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": ()},
+            },
+            {
+                "total_tlu_count": 1,
+            },
+        ),
+        pytest.param(
+            lambda x: x**2,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": (3,)},
+            },
+            {
+                "total_tlu_count": 3,
+            },
+        ),
+        pytest.param(
+            lambda x: x**2,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": (3, 2)},
+            },
+            {
+                "total_tlu_count": 3 * 2,
+            },
+        ),
+        pytest.param(
+            lambda x, y: x * y,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": ()},
+                "y": {"status": "encrypted", "range": [0, 10], "shape": ()},
+            },
+            {
+                "total_tlu_count": 2,
+            },
+        ),
+        pytest.param(
+            lambda x, y: x * y,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": (3,)},
+                "y": {"status": "encrypted", "range": [0, 10], "shape": (3,)},
+            },
+            {
+                "total_tlu_count": 3 * 2,
+            },
+        ),
+        pytest.param(
+            lambda x, y: x * y,
+            {
+                "x": {"status": "encrypted", "range": [0, 10], "shape": (3, 2)},
+                "y": {"status": "encrypted", "range": [0, 10], "shape": (3, 2)},
+            },
+            {
+                "total_tlu_count": 3 * 2 * 2,
+            },
+        ),
+    ],
+)
+def test_statistics(function, parameters, expected_statistics, helpers):
+    """
+    Test statistics of the circuit provided by the compiler.
+    """
+
+    parameter_encryption_statuses = helpers.generate_encryption_statuses(parameters)
+    configuration = helpers.configuration()
+
+    compiler = fhe.Compiler(function, parameter_encryption_statuses)
+
+    inputset = helpers.generate_inputset(parameters)
+    circuit = compiler.compile(inputset, configuration)
+
+    for name, expected_value in expected_statistics.items():
+        assert hasattr(circuit, name)
+        assert getattr(circuit, name) == expected_value
```

Comment:
to me, the best way to realize what this feature would be about, and notably to fine-tune the statistics which are needed for ML or users in general, would be:
- take a larger MLIR, maybe a typical one from ML
- compile it, and compare its statistics with an expected statistics str

Maybe some of you will claim it's not good since the expected str will change often. What I think is that:
- when it changes for good reason, the expected str can be easily updated
- and this later will find regressions which would have been hard to see, like if we had PBS for a bad reason
- that kind of test is an excellent way to describe what the feature does 

Remark that the test could also be in the `pytest_codeblocks` if you prefer

Reply:
I have not understood what you mean. How could something be changed in CML by some APIs they don't call for now?

Reply:
I'm not saying now, I'm saying big things that are frequently updated should be tested where they are defined. So, once we merge this PR and release it, ML team should introduce their own big tests for this purpose.

Reply:
It's tested in CP, but on primitive or small functions. If they work properly, it can be generalized and that's what we need to test in CP.